stl/inc/algorithm

```diff
@@ -5634,9 +5634,82 @@ _NODISCARD bool is_sorted(_ExPo&& _Exec, _FwdIt _First, _FwdIt _Last) noexcept /
     return _STD is_sorted(_STD forward<_ExPo>(_Exec), _First, _Last, less{});
 }
 
-#endif // _HAS_CXX17
+#ifdef __cpp_lib_concepts
+namespace ranges {
+    // FUNCTION TEMPLATE _Is_sorted_until_unchecked
+    template <class _It, class _Se, class _Pr, class _Pj>
+    _NODISCARD constexpr _It _Is_sorted_until_unchecked(_It _First, const _Se _Last, _Pr _Pred, _Pj _Proj) {
+        _STL_INTERNAL_STATIC_ASSERT(forward_iterator<_It>);
+        _STL_INTERNAL_STATIC_ASSERT(sentinel_for<_Se, _It>);
+        _STL_INTERNAL_STATIC_ASSERT(indirect_strict_weak_order<_Pr, projected<_It, _Pj>>);
+
+        if (_First != _Last) {

```

Comment:
To be consistent in my bikeshedding. Early return?

Reply:
I didn't early-return here because adding two lines to save 5 indents seems like a wash to me. I'll change it if you have strong feelings and/or other reviewers agree.

Reply:
Nah, I was just severely burned at my last job by inheriting a codebase where double digit indentation was the norm.

That said, I really value consistency within the larger codebase. So if we use early returns in some functions and not in others that somehow bothers me as I generally search for the reason why it is different this time. Almost all algorithms have the `_First == _Last` check so treating the same way everywhere does make sense IMHO.

Reply:
Consistency is a good reason, and early return is generally desirable, even in fairly short functions like this. Reducing indentation is nice, but the real simplification is that we can read the code and say "ok, we've definitively dealt with the empty case, now let's assume non-empty" and that's worth replicating the (trivial) return statement. If the main algorithm were a *single* line, I think early return would probably be self-defeating, but here I'd be mildly in favor of it.

(I appreciate your review comments - they make us think about things that we wouldn't necessarily think of otherwise. In this case, I'd probably have been lazy and not commented about the inconsistency. Thanks!)

Reply:
> Nah, I was just severely burned at my last job by inheriting a codebase where double digit indentation was the norm.
> 
> That said, I really value consistency within the larger codebase. So if we use early returns in some functions and not in others that somehow bothers me as I generally search for the reason why it is different this time. Almost all algorithms have the `_First == _Last` check so treating the same way everywhere does make sense IMHO.

You had me at "I really value consistency within the larger codebase" ;)